### DISCUSSION TITLE
Assign labels to ingredients on recipe screen / update ingredients with labels added in list

### DISCUSSION THREAD

**Discussion by jack5mikemotown** [mealie-recipes/mealie#2393] [Feature Request]
I really like the ability to throw all of the ingredients from a recipe into a list, but one thing I find is that the current way to add labels to ingredients isn't really intuitive, as you have to dig into the settings to modify each label data table. If you add them from the list section, they don't save or apply to the actual ingredients. And when you create an ingredient from the recipe page, it doesn't give you the option to add a label.

Two different solutions that would really help in adding labels to ingredients would be to either allow you to assign a label to an ingredient on the recipe screen, or have the labels assigned from a list carryover to the entry for the ingredient. 

Either of these suggestions would definitely make the process of adding ingredients to a list much more seamless

**Comment by madhathacker** [comment]
I agree with this feature request. It is similar to https://github.com/mealie-recipes/mealie/discussions/2615. 

The simplest solution would be to add an "label" column to each ingredient entry in a recipe. Here is an example represented in JSON below:
```json
{
    "quantity": 1,
    "unit": null,
    "food": null,
    "note": "4 6-oz. salmon fillets, patted dry with a paper towel",
    "isFood": false,
    "disableAmount": true,
    "display": "4 6-oz. salmon fillets, patted dry with a paper towel",
    "title": null,
    "originalText": null,
    "referenceId": "11d26293-7f9b-4202-ab8b-13e6e2ad6914",
** "label": "fish" **
},
```
After storing the extra information in the recipe data structure, we would simply update the "Send to Shopping List" function to grab the label of the ingredients along with the names and quantities.

The more complicated solutions would be to add an ingredient catalog that keeps track of all ingredients and their properties which are shared across all recipes. But this would have a lot of edge cases and added complexity. For example "canned tomatoes" in one recipe might be very different from "canned tomatoes" in another.

**Reply by michael-genson** [reply]
I'm confused, ingredients are already automatically labeled based on their food setting. As long as your foods have labels then when adding ingredients to a shopping list they automatically have labels

**Reply by michael-genson** [reply]
More info in the FAQ about foods and data management: https://docs.mealie.io/documentation/getting-started/faq/

**Reply by madhathacker** [reply]
Woah. Me and my friend just spend the last 3 hours playing around with lots of the features on the FAQ. We didn't realize you could parse the imported recipes to pull out the food items. It wasn't very intuitive as in order to find and manage the food items you have to navigate through your own user profile. Also, none of the default food items had labels, so I didn't realize why things weren't working how I expected them to. 

You have done a great job implementing a TON of amazing stuff! Hats of to you! 

I think the interface just needs a bit of work to help make things a bit friendlier for new users. As another example, I couldn't figure out why the recipe finder wasn't bringing up any of my imported recipes when I first started. It turned out I hadn't actually parsed the food items out.

I think I will spend a bit of time adding and labeling food items in my instance. I might try using LLMs to speed the processes up.

**Reply by madhathacker** [reply]
Me and my friend are loving this project so far; I would love to help out with some contributions where I can. I will be looking for your Contributors Guide tomorrow.

**Reply by michael-genson** [reply]
Glad to hear! Yes there's definitely a lot of work that can be done to make parsing/data management easier and more intuitive.

For new instances, there is a PR open to make the seed data vastly more useful by attaching labels to foods automatically: https://github.com/mealie-recipes/mealie/pull/4812
